src/Scientist/OperationException.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace GitHub
+{
+    /// <summary>
+    /// Thrown when an exception was thrown during the execution of an operation.
+    /// </summary>
+    public class OperationException : Exception
+    {
+        /// <summary>
+        /// The operation in which this exception occurred within.
+        /// </summary>
+        public Operation Operation { get; }
+
+        public override string Message { get; }
+
+        internal OperationException(Operation operation, Exception inner) : base(null, inner)
```

Comment:
Should we use the base class constructor to assign the exception's message here, i.e., `base(GetExceptionMessage(operation), inner)`?